packages/contracts/src/utils/exchange_wrapper.ts

```diff
@@ -142,12 +172,30 @@ export class ExchangeWrapper {
     ): Promise<TransactionReceiptWithDecodedLogs> {
         const params = formatters.createMarketFillOrders(orders, opts.takerTokenFillAmount);
         const txHash = await this._exchange.marketFillOrders.sendTransactionAsync(
-            params.orderAddresses,
-            params.orderValues,
+            params.orders,
             params.takerTokenFillAmount,
-            params.v,
-            params.r,
-            params.s,
+            params.signatures,
+            { from },
+        );
+        const tx = await this._zeroEx.awaitTransactionMinedAsync(txHash);
+        tx.logs = _.filter(tx.logs, log => log.address === this._exchange.address);
+        tx.logs = _.map(tx.logs, log => {
+            const logWithDecodedArgs = this._logDecoder.decodeLogOrThrow(log);
+            wrapLogBigNumbers(logWithDecodedArgs);
+            return logWithDecodedArgs;
+        });
```

Comment:
These 6 lines are repeated many times in this file. Move them to a helper function to keep the code DRY.